pystiche/papers/sanakoyeu_et_al_2018/modules.py

```diff
@@ -0,0 +1,435 @@
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=1,
+            padding="valid",
+            use_act=False,
+        ),
+        nn.ReflectionPad2d(padding),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=1,
+            padding="valid",
+            use_act=False,
+        ),
+    )
+
+    return ResidualBlock(residual)
+
+
+def sanakoyeu_et_al_2018_transformer_encoder(
+    in_channels: int = 3,
+) -> pystiche.SequentialModule:
+    modules = (
+        nn.ReflectionPad2d(15),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels=in_channels,
+            out_channels=32,
+            kernel_size=3,
+            stride=1,
+            padding="valid",
+        ),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels=32, out_channels=32, kernel_size=3, stride=2, padding="valid",
+        ),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels=32, out_channels=64, kernel_size=3, stride=2, padding="valid",
+        ),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels=64, out_channels=128, kernel_size=3, stride=2, padding="valid",
+        ),
+        SanakoyeuEtAl2018ConvBlock(
+            in_channels=128, out_channels=256, kernel_size=3, stride=2, padding="valid",
+        ),
+    )
+    return SequentialEncoder(modules)
+
+
+def sanakoyeu_et_al_2018_transformer_decoder(
+    out_channels: int = 3, num_res_block: int = 9,
+) -> pystiche.SequentialModule:
+
+    modules = []
+    for i in range(num_res_block):
+        modules.append(sanakoyeu_et_al_2018_residual_block(256))
+
+    modules.append(
+        SanakoyeuEtAl2018ConvTransponseBlock(
+            in_channels=256, out_channels=256, kernel_size=3, stride=2, padding="same",
+        )
+    )
+    modules.append(
+        SanakoyeuEtAl2018ConvTransponseBlock(
+            in_channels=256, out_channels=128, kernel_size=3, stride=2, padding="same",
+        )
+    )
+    modules.append(
+        SanakoyeuEtAl2018ConvTransponseBlock(
+            in_channels=128, out_channels=64, kernel_size=3, stride=2, padding="same",
+        )
+    )
+    modules.append(
+        SanakoyeuEtAl2018ConvTransponseBlock(
+            in_channels=64, out_channels=32, kernel_size=3, stride=2, padding="same",
+        )
+    )
+    modules.append(nn.ReflectionPad2d(3))
+    modules.append(
+        sanakoyeu_et_al_2018_conv(
+            in_channels=32,
+            out_channels=out_channels,
+            kernel_size=7,
+            stride=1,
+            padding="valid",
+        )
+    )
+    return pystiche.SequentialModule(*modules)
+
+
+class SanakoyeuEtAl2018Decoder(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.decoder = sanakoyeu_et_al_2018_transformer_decoder()
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        return torch.sigmoid(self.decoder(input)) * 2 - 1
```

Comment:
We should create an `nn.Module` for this an just drop it as a last layer into the `decoder`. Otherwise this is not shown when you print this.